test/com/ableton/JenkinsMocks.groovy

```diff
@@ -17,4 +17,41 @@ class JenkinsMocks {
       return System.properties['user.dir']
     }
   }
+
+  static Closure sh = { args ->
+    String script = null
+    def returnStdout = false
+    def returnStatus = false
+
+    // Default argument to sh command
+    if (args instanceof String || args instanceof GString) {
+      script = args
+    } else if (args instanceof Map) {
+      script = args['script']
+      returnStatus = args['returnStatus'] ?: false
+      returnStdout = args['returnStdout'] ?: false
+      if (returnStatus && returnStdout) {
+        throw new IllegalArgumentException(
+          'returnStatus and returnStdout are mutually exclusive options')
+      }
+    }
+    assert script
+
+    Process process = script.execute()
```

Comment:
doesn't sound like a mock... sounds like it's actually running what we tell it to?

Reply:
unless you're saying the override in https://github.com/AbletonDevTools/python-pipeline-utils/pull/6/commits/9359777093ff166454fe8ac710983db7554d763f applies here?

Reply:
@mst-ableton No, that's correct... it's actually executing the script. It is possible to mock the output of `sh` by returning a string, but I found this to be rather clumsy especially when functions which contained multiple `sh` calls. I guess it's less of a "mock" and more of an "implementation" then. 😅 

I guess we could theoretically mock `sh` in a way that we pass it a map of key/value pairs corresponding to the arguments/mocked return value, but I don't like the direction that is heading in, since it would involve a lot of extra hacks in the `setUp` function.

Reply:
The key/value pairs are what I typically do, it does mean extra work but it also means the tests can work on any OS.

I don't want to block this PR because of that, but we should at least make a comment here describing that the mock is only a semi-mock.

Reply:
Nah, I'd rather get it right the first try rather than procrastinate with a `// TODO`. 😉 I think I have a good way to implement this, I'll push a fixup soon.